EIPS/eip-4844.md

```diff
@@ -146,16 +147,18 @@ def evaluate_polynomial_in_evaluation_form(poly: List[BLSFieldElement], x: BLSFi
     return r
 ```
 
-Approximates `2 ** (numerator / denominator)`, with the simplest possible approximation that is continuous and has a continuous derivative:
+Approximates `factor * e ** (numerator / denominator)` using Taylor expansion:
 
 ```python
-def fake_exponential(numerator: int, denominator: int) -> int:
-    cofactor = 2 ** (numerator // denominator)
-    fractional = numerator % denominator
-    return cofactor + (
-        fractional * cofactor * 2 +
-        (fractional ** 2 * cofactor) // denominator
-    ) // (denominator * 3)
+def fake_exponential(factor: int, numerator: int, denominator: int) -> int:
```

Comment:
Is there a rationale for switching to the Taylor expansion?

Reply:
This came out of a discussion with @vbuterin. If I recall correctly, reasons were:

- simpler conceptually
- more precise approximation
- only slightly more computationally expensive (and only done once per block, so insignificant)

Reply:
I think this is "conceptually simpler" for a *very* small set of people (math lovers).  I think for the vast majority of the rest of the world (which likely includes most of the people building Ethereum and building stuff on Ethereum) the original was "conceptually simpler".

I don't think the precision here is necessary, given that we are just guessing at capacity targets anyway.

Reply:
Conceptually simpler not from a "reading the code", but an "understanding where the code came from" perspective. Which I would argue is the main objective.